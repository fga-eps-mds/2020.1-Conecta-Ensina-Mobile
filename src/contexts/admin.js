import React, {createContext, useState, useContext, useEffect} from 'react';
import {AuthContext} from './auth';

export const AdmContext = createContext({});

function AdmProvider({children}) {
  const [students, setStudents] = useState(null);
  const [userDB, setUserDB] = useState(null);
  const [teacher, setTeacher] = useState(null);

  const {Host, user} = useContext(AuthContext);

  async function getProfessorList() {
    const fetchResponse = await fetch(`${Host}/api/student/status/0`);

    console.log(Host);

    try {
      const data = await fetchResponse.json();
      setStudents(data.data.student);
    } catch (error) {
      return error;
    }
  }

  async function statusUpdate(id) {
    const settings = {
      method: 'PUT',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        status: 1,
        agentRole: user.role,
      }),
    };
    const fetchResponse1 = await fetch(
      `${Host}/api/student/status/${id}`,
      settings,
    );
    try {
      const data = await fetchResponse1.json();
      console.log('Success:', data);
    } catch (error) {
      console.error('Error:', error);
    }
  }

  async function getProfessoUser(id) {
    const fetchResponse = await fetch(`${Host}/api/user/${id}`);
    try {
      const data = await fetchResponse.json();
      let usuario = {
        id: id,
        firstName: data.data.user.firstName,
        lastName: data.data.user.lastName,
        email: data.data.user.email,
        password: data.data.user.password,
        role: data.data.user.role,
        cellphone: data.data.user.cellphone,
      };
      setUserDB(usuario);
      try {
        let response = await fetch(`${Host}/api/teacher/${id}`);
        const data = await response.json();
        let professor = {
          photo: data.data.teacher.photo,
          video: data.data.teacher.video,
          graduation_area: data.data.teacher.graduation_area,
          degree: data.data.teacher.degree,
          bank: data.data.teacher.bank,
          agency: data.data.teacher.agency,
          account: data.data.teacher.account,
        };
        setTeacher(professor);
      } catch (error) {
        console.error(error);
      }
    } catch (error) {
      return error;
    }
  }

  return (
    <AdmContext.Provider
      value={{
        students,
        userDB,
        teacher,
        getProfessorList,
        statusUpdate,
        getProfessoUser,
      }}>
      {children}
    </AdmContext.Provider>
  );
}

export default AdmProvider;
