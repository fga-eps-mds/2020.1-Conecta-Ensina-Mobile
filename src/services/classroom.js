export async function getNextClassroom(Host) {
  const fetchResponse = await fetch(
    `${Host}/api/classroom/nextClass/3bd7c190-ce64-4827-8c0c-58cfef45ad9f`,
  );
  try {
    const data = await fetchResponse.json();
    return data.data.classroom;
  } catch (error) {
    return error;
  }
}

export async function getUserClassroom(Host, id) {
  const fetchResponse = await fetch(`${Host}/api/classroom/userClasses/${id}`);
  try {
    const data = await fetchResponse.json();
    return data.data.classroom;
  } catch (error) {
    return error;
  }
}

export async function createClass(user, teacher, filter, student, Host) {
  const response = await fetch(`${Host}/api/classroom/create`, {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      agentRole: user.role,
      teacher: teacher,
      student: user.id,
      grade: student.grade,
      subject: filter.subject,
      dtclass: filter.dtClass,
      duration: filter.duration,
      cep: student.cep,
      number: student.number,
      details: 'oi',
    }),
  });
  try {
    const data = await response.json();
    console.log('Success: ', data);
    return data;
  } catch (error) {
    console.log(error);
  }

  console.log(response);
}
