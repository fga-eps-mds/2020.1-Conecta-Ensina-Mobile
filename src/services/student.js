export async function getStudent(Host, id) {
  const response = await fetch(`${Host}/api/student/${id}`);
  try {
    const data = await response.json();

    //setSubjects(data.data.subject);
    return data.data;
  } catch (error) {
    return error;
  }
}