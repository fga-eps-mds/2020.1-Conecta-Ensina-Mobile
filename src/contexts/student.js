import React, {createContext, useState, useContext} from 'react';
import {AuthContext} from './auth';
import * as Student from '../services/student';

export const StudentContext = createContext({});

export default function UserProvider({children}) {
  const [student, setStudent] = useState({});
  const {Host} = useContext(AuthContext);

  async function getStudent(id) {
    const response = await Student.getStudent(Host, id);
    if (student !== response) {
      console.log(response);
      setStudent(response);
    }
  }

  return (
    <StudentContext.Provider value={{student, getStudent}}>
      {children}
    </StudentContext.Provider>
  );
}
