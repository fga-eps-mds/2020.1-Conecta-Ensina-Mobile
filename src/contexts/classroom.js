import React, {createContext, useState, useContext} from 'react';
import {AuthContext} from './auth';
import * as Class from '../services/classroom';
import * as Student from '../services/student';
import * as Address from '../services/findAddress';
import * as GeoCode from '../services/geocode';
import {FiltersContext} from '../contexts/filters';
import {StudentContext} from '../contexts/student';

export const ClassroomContext = createContext({});

export default function ClassroomProvider({children}) {
  const [classroom, setClassroom] = useState({});
  const [firstClass, setFirstClass] = useState(null);
  const [classes, setClasses] = useState(null);
  const {Host, user} = useContext(AuthContext);
  const {stackStudents} = useContext(StudentContext);
  const {filter} = useContext(FiltersContext);
  const [statusClasses, setStatusClasses] = useState([]);
  const [statusClass, setStatusClass] = useState({});
  const [coordenates, setCoordenates] = useState({});

  async function loadNextClass() {
    const response = await Class.getNextClassroom(Host);
    if (classroom !== response) {
      setFirstClass(response);
    }
  }
  async function createClass(teacher) {
    //console.log('Esse aqui Ã© o professor: ' + teacher);
    const student = await Student.getStudent(Host, user.id);
    const response = await Class.createClass(
      user,
      teacher,
      filter,
      student.student,
      Host,
    );
  }

  async function loadUserClasses() {
    const response = await Class.getUserClassroom(Host, user.id);
    if (classroom !== response) {
      setClassroom(response);
    }
  }

  async function getClassroom(status) {
    let teacher, student;
    if (user.role === 2) {
      student = '';
      teacher = user.id;
    } else {
      student = user.id;
      teacher = '';
    }
    const response = await Class.getClassroom(student, teacher, status, Host);

    if (classroom !== response) {
      //console.log(response);
      setStatusClasses(response);
    }
  }

  async function updateStatusClassroom(id) {
    let response = await Class.readClass(Host, id);
    //console.log(response.status);
    if (response.status === 1) {
      response = await Class.updateStatusClassroom(id, 2, Host);
      //console.log(response.message);
    } else if (response.status === 2) {
      response = await Class.updateStatusClassroom(id, 3, Host);
      //console.log(response.message);
    } else if (response.status === 3) {
      response = await Class.updateStatusClassroom(id, 4, Host);
      //console.log(response.message);
    } else if (response.status === 4) {
      response = await Class.updateStatusClassroom(id, 5, Host);
      //console.log(response.message);
    }
  }
  async function loadStatusClasses(status) {
    const response = await Class.getStatusClassroom(Host, user.id, status);
    if (statusClasses !== response) {
      await setStatusClasses(response);
    }
    //console.log(response);
    return response;
  }

  async function loadStatusClassesStudents(status) {
    const response = await loadStatusClasses(status);
    let studentList = await {list: []};
    await Promise.all(
      response.map(async (item) => {
        const student = await Student.getStudent(Host, item.student);
        studentList.list = await studentList.list.concat(student);
      }),
    );
    await stackStudents(studentList.list);
  }

  async function readClass(id) {
    const response = await Class.readClass(Host, id);
    const address = await Address.findAddress(response.cep);
    let responseClass = {
      id: response.role,
      status: response.status,
      teacher: response.teacher,
      student: response.student,
      grade: response.grade,
      subject: response.subject,
      dtclass: response.dtClass,
      duration: response.duration,
      cep: response.cep,
      number: response.number,
      details: response.details,
      address: address,
    };

    setClassroom(responseClass);
  }

  async function getClass() {
    const response = await Class.getClass(Host, user);
    setClasses(response);
  }

  async function updateStatusClasses(id, status) {
    const response = await Class.updateStatus(id, status, Host);
    if (classroom !== response) {
      setClassroom(response);
    }
  }

  async function geoCode(cep) {
    const response = await GeoCode.geocode(cep);
    if (coordenates !== response) {
      //console.log(response.results[0].geometry.location);
      setCoordenates(response.results[0].geometry.location);
    }
  }

  return (
    <ClassroomContext.Provider
      value={{
        classroom,
        firstClass,
        classes,
        loadNextClass,
        createClass,
        loadUserClasses,
        getClassroom,
        updateStatusClassroom,
        loadStatusClasses,
        readClass,
        statusClasses,
        statusClass,
        setStatusClass,
        setStatusClasses,
        Host,
        getClass,
        updateStatusClasses,
        loadStatusClassesStudents,
        geoCode,
        coordenates,
      }}>
      {children}
    </ClassroomContext.Provider>
  );
}
