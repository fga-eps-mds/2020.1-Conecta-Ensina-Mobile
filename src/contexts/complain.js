import React, {createContext, useState, useContext} from 'react';
import {AuthContext} from './auth';
import * as Student from '../services/student';

export const ComplainContext = createContext({});

export default function ComplainProvider({children}) {
  const [complain, setComplain] = useState({});
  const {Host, user} = useContext(AuthContext);

  async function createComplain(details,accused) {
    const settings = {
      method: 'POST',
      headers: {
        Accept: 'application/json',
        'Content-Type':'application/json',
      },
      body:JSON.stringify({
        accused: accused,
        reported_by: user.id,
        details: details,
      })
    }
    const response = await fetch(`${Host}/api/complain/create`, settings)
    try{
      const data = response.json();
      console.log("Complain:", data);
      if (data.message === 'Den√∫ncia criada com sucesso!'){
        return data;
      }
    } catch(error) {
      console.error("error: ", error)
    }
  }

  return (
    <ComplainContext.Provider value={{complain, createComplain}}>
      {children}
    </ComplainContext.Provider>
  );
}