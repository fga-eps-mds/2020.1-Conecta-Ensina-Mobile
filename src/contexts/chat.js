import React, {createContext, useState, useContext, useEffect} from 'react';
import {AuthContext} from './auth';
import * as Chat from '../services/chat';
import { ClassroomContext } from './classroom';

export const chatContext = createContext({});

export default function ChatProvider({children}) {
  const {Host, user} = useContext(AuthContext);
  const {classroom} = useContext(ClassroomContext);
  const [chat, setChat] = useState(null);

  useEffect(() => {
      readChat()
  }, [])

  async function createChat(text) {
    const response = await Chat.createMessage(
        user.id, 
        text, 
        classroom.id, 
        classroom.teacher, 
        classroom.student, 
        Host)
  }

  async function readChat() {
    console.log(classroom)
    const response = await Chat.getChat(Host, classroom.id);
    if (chat !== response) {
      setChat(response);
    }
  }
  return (
    <chatContext.Provider
      value={{
        readChat,
        createChat,
        chat
      }}>
      {children}
    </chatContext.Provider>
  );
}
