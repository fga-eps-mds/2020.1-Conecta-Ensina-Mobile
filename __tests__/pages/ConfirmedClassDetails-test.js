import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import ConfirmedClassDetais from '../../src/pages/ConfirmedClassDetails';
import {ClassroomContext} from '../../src/contexts/classroom';
import {StudentContext} from '../../src/contexts/student';

jest.mock('../../src/services/dateResolver');
jest.mock('../../src/services/timeResolver');

describe('Testing ConfirmedClassDetais page', () => {
  const navigation = {
    navigate: jest.fn(),
  };
  const statusClass = {
    subject: 'subject',
    dtclass: 11,
    duration: 11,
  };
  const student = {
    user: {firstName: 'first', lastName: 'last'},
    student: {grade: 'grade'},
  };

  test('Should render page', () => {
    const tree = render(
      <StudentContext.Provider value={{student}}>
        <ClassroomContext.Provider value={{statusClass}}>
          <ConfirmedClassDetais navigation={navigation} />
        </ClassroomContext.Provider>
      </StudentContext.Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should press ContinueContainer button', () => {
    const {getByTestId} = render(
      <StudentContext.Provider value={{student}}>
        <ClassroomContext.Provider value={{statusClass}}>
          <ConfirmedClassDetais navigation={navigation} />
        </ClassroomContext.Provider>
      </StudentContext.Provider>,
    );

    const continueContainer = getByTestId('ContinueContainer');
    fireEvent.press(continueContainer);
    expect(navigation.navigate).toBeCalled();
  });
});
