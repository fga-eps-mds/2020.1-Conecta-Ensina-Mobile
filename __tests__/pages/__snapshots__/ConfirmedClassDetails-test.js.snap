// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ConfirmedClassDetais page Should render page 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#184E79",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
  testID="background"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#184E79",
          "flexBasis": 0,
          "flexGrow": 0.7,
          "flexShrink": 1,
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "marginBottom": "8.4%",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/logo.png",
          }
        }
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 2,
              "flexShrink": 1,
              "height": 120,
              "width": 120,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderRadius": 100,
              "flexBasis": 0,
              "flexGrow": 1.2,
              "flexShrink": 1,
              "height": 58,
              "justifyContent": "center",
              "marginTop": "5.5%",
              "width": 58,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/user_blue.png",
            }
          }
          style={
            Array [
              Object {
                "height": "63%",
                "width": "52%",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 0.8,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginTop": "5%",
              "paddingTop": "1%",
            },
          ]
        }
      >
        <View
          marginTop={
            Object {
              "value": "2%",
            }
          }
          style={
            Array [
              Object {
                "marginBottom": 0,
                "marginTop": "2%",
              },
            ]
          }
        >
          <Text
            bigMedium={true}
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 30,
                  "textAlign": "center",
                },
              ]
            }
            white={true}
          />
        </View>
        <View
          marginTop={
            Object {
              "value": "2%",
            }
          }
          style={
            Array [
              Object {
                "marginBottom": 0,
                "marginTop": "2%",
              },
            ]
          }
        >
          <Text
            smallMedium={true}
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                  "textAlign": "center",
                },
              ]
            }
            white={true}
          >
            Invalid Grade
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexDirection": "column",
              "flexGrow": 0.8,
              "flexShrink": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          marginBot={
            Object {
              "value": "1%",
            }
          }
          marginTop={
            Object {
              "value": "2%",
            }
          }
          style={
            Array [
              Object {
                "marginBottom": "1%",
                "marginTop": "2%",
              },
            ]
          }
        >
          <Text
            black={true}
            smallMedium={true}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                  "textAlign": "center",
                },
              ]
            }
          >
            Disciplina
          </Text>
        </View>
        <View
          medium={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#E46270",
                "borderRadius": 15,
                "height": 40,
                "justifyContent": "center",
                "textAlign": "center",
                "width": 196,
              },
            ]
          }
        >
          <Text
            small={true}
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 14,
                  "textAlign": "center",
                },
              ]
            }
            white={true}
          >
            subject
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexDirection": "column",
              "flexGrow": 0.8,
              "flexShrink": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          marginBot={
            Object {
              "value": "1%",
            }
          }
          marginTop={
            Object {
              "value": "2%",
            }
          }
          style={
            Array [
              Object {
                "marginBottom": "1%",
                "marginTop": "2%",
              },
            ]
          }
        >
          <Text
            black={true}
            smallMedium={true}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                  "textAlign": "center",
                },
              ]
            }
          >
            Data
          </Text>
        </View>
        <View
          medium={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#E46270",
                "borderRadius": 15,
                "height": 40,
                "justifyContent": "center",
                "textAlign": "center",
                "width": 196,
              },
            ]
          }
        >
          <Text
            small={true}
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 14,
                  "textAlign": "center",
                },
              ]
            }
            white={true}
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexDirection": "column",
              "flexGrow": 0.8,
              "flexShrink": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          marginBot={
            Object {
              "value": "1%",
            }
          }
          marginTop={
            Object {
              "value": "2%",
            }
          }
          style={
            Array [
              Object {
                "marginBottom": "1%",
                "marginTop": "2%",
              },
            ]
          }
        >
          <Text
            black={true}
            smallMedium={true}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                  "textAlign": "center",
                },
              ]
            }
          >
            Hor√°rio
          </Text>
        </View>
        <View
          medium={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#E46270",
                "borderRadius": 15,
                "height": 40,
                "justifyContent": "center",
                "textAlign": "center",
                "width": 196,
              },
            ]
          }
        >
          <Text
            small={true}
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 14,
                  "textAlign": "center",
                },
              ]
            }
            white={true}
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexDirection": "column",
              "flexGrow": 0.8,
              "flexShrink": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          marginBot={
            Object {
              "value": "1%",
            }
          }
          marginTop={
            Object {
              "value": "2%",
            }
          }
          style={
            Array [
              Object {
                "marginBottom": "1%",
                "marginTop": "2%",
              },
            ]
          }
        >
          <Text
            black={true}
            smallMedium={true}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                  "textAlign": "center",
                },
              ]
            }
          >
            Dura√ß√£o
          </Text>
        </View>
        <View
          medium={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#E46270",
                "borderRadius": 15,
                "height": 40,
                "justifyContent": "center",
                "textAlign": "center",
                "width": 196,
              },
            ]
          }
        >
          <Text
            small={true}
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 14,
                  "textAlign": "center",
                },
              ]
            }
            white={true}
          >
            11 Horas
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexDirection": "column",
              "flexGrow": 0.8,
              "flexShrink": 1,
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          marginBot={
            Object {
              "value": "1%",
            }
          }
          marginTop={
            Object {
              "value": "2%",
            }
          }
          style={
            Array [
              Object {
                "marginBottom": "1%",
                "marginTop": "2%",
              },
            ]
          }
        >
          <Text
            black={true}
            smallMedium={true}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontFamily": "Roboto",
                  "fontSize": 20,
                  "textAlign": "center",
                },
              ]
            }
          >
            Modalidade
          </Text>
        </View>
        <View
          medium={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#E46270",
                "borderRadius": 15,
                "height": 40,
                "justifyContent": "center",
                "textAlign": "center",
                "width": 196,
              },
            ]
          }
        >
          <Text
            small={true}
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 14,
                  "textAlign": "center",
                },
              ]
            }
            white={true}
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "marginTop": 0,
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#297FB8",
              "borderRadius": 8,
              "height": 31,
              "justifyContent": "center",
              "opacity": 1,
              "width": 138,
            }
          }
          testID="ContinueContainer"
        >
          <Text
            bigSmall={true}
            style={
              Array [
                Object {
                  "color": "#ffffff",
                  "fontFamily": "Roboto",
                  "fontSize": 16,
                  "textAlign": "center",
                },
              ]
            }
            white={true}
          >
            Continuar
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
